.github/workflows/deploy.yaml

```diff
@@ -49,4 +76,138 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: type=gha
           cache-to: type=gha,mode=max
+          # Fix incorrect container type sidebar display in GitHub Container
+          # Registry
           provenance: false
+
+      - name: Save image name to output
+        id: save-image-name
+        run: |
+          IMAGE_NAME=$(echo "$METADATA" | jq -r '."image.name"')
+          echo "image-name=$IMAGE_NAME" >> "$GITHUB_OUTPUT"
+        shell: bash
+        env:
+          METADATA: ${{ steps.build-and-push.outputs.metadata }}
+
+  run-model:
+    needs: [publish-docker-image]
+    runs-on: ubuntu-latest
+    # Require manual approval to run this job
+    environment: deploy
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint
+    # so that we can authenticate with AWS
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Terraform
+        uses: ./.github/actions/setup-terraform
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+          aws-account-id: ${{ secrets.AWS_ACCOUNT_ID }}
+          batch-container-image-name: ${{ needs.publish-docker-image.outputs.image-name }}
+          role-duration-seconds: 14400  # Worst-case time for a full model run
```

Comment:
Does 4 hours seem like a reasonable timeout to set on this task?

Reply:
Cool, I'll plan to leave this as-is and we can adjust as we do performance improvements then.